### ISSUE TITLE
issues running cri-o with Clear Containers

### ISSUE THREAD

**Issue by runcom** [kubernetes-incubator/cri-o#332]
Basically followed https://github.com/01org/cc-oci-runtime/wiki/Installing-Clear-Containers-on-Fedora-24 adapting them to F25 (since there's already an F25 repo). I can start clear containers with docker 1.12.1 but when it comes to try it with cri-o I'm getting:
```
$ sudo /home/amurdaca/go/bin/ocid --debug --log ocid.log --runtime /usr/bin/cc-oci-runtime --conmon $PWD/conmon/conmon --seccomp-profile=$PWD/seccomp.json
WARN[0000] default configuration file does not exist: /etc/ocid/ocid.conf
container PID: 4799
qemu-lite-system-x86_64: -chardev pipe,id=charconsole0,path=/run/containers/devicemapper-containers/b2245cd26b8ac53e9118fa291e04ff29f488b6f85a89d83e063a98d11f535a01/userdata/cc-std: Could not open '/run/containers/devicemapper-containers/b2245cd26b8ac53e9118fa291e04ff29f488b6f85a89d83e063a98d11f535a01/userdata/cc-std': No such file or directory
PID 4799 exited with status 1
conmon: Failed to create container
conmon: Failed to create container
```
and in kubernetes:
```
Events:
  FirstSeen     LastSeen        Count   From                    SubObjectPath           Type            Reason          Message
  ---------     --------        -----   ----                    -------------           --------        ------          -------
  3m            3m              1       {default-scheduler }                            Normal          Scheduled       Successfully assigned nginx-2932712833-s75ld to 127.0.0.1
  3m            3m              1       {kubelet 127.0.0.1}                             Normal          SandboxReceived Pod sandbox received, it will be created.
  2m            19s             3       {kubelet 127.0.0.1}     spec.containers{nginx}  Normal          Pulled          Successfully pulled image "nginx"
  2m            19s             3       {kubelet 127.0.0.1}     spec.containers{nginx}  Warning         Failed          Failed to create container with error: rpc error: code = 2 desc = reading pid from init pipe: EOF
  2m            19s             3       {kubelet 127.0.0.1}                             Warning         FailedSync      Error syncing pod, skipping: failed to "StartContainer" for "nginx" with rpc error: code = 2 desc = reading pid from init pipe: EOF: "Create Container Failed"

  3m    18s     4       {kubelet 127.0.0.1}     spec.containers{nginx}  Normal  Pulling pulling image "nginx"
```
/cc @sameo @jjlakis

**Comment by sameo** [comment]
@runcom The [Clear Containers instructions](https://github.com/01org/cc-oci-runtime/wiki/Installing-Clear-Containers-on-Fedora-24) you looked at are for Clear Containers 2.0 and that will not work with CRI-O.

Some background here, as we're currently transitioning to Clear Containers 2.1:

Clear Containers 2.0 was a (too) simple architecture that did not allow us to support many Docker commands and certainly not anything related to pods, k8s or CRI-O. Clear Containers is a partial re-architecture of 2.0 to fix most of those gaps.
The Clear Containers [master branch](https://github.com/01org/cc-oci-runtime) is currently running the 2.1 architecture and I released 2.1.0-rc.4 yesterday. I test CRI-O and Clear Containers with master only. We expect to do the final 2.1.0 release in 3-4 weeks. Once this is done we'll update the installation instructions.
I'll be happy to provide more details about all of this during next week meeting.

Also, I will prepare a temporary wiki page with the 2.1 installation instructions for you to look at. We already are building OBS packages for 2.1 on FC25 in case you want to try them out:
http://download.opensuse.org/repositories/home:/clearlinux:/preview:/clear-containers-2.1/Fedora_25/

**Comment by runcom** [comment]
@sameo thanks, however, using the new packages is not helping:
```
Events:
  FirstSeen     LastSeen        Count   From                    SubObjectPath   Type            Reason          Message
  ---------     --------        -----   ----                    -------------   --------        ------          -------
  34s           34s             1       {default-scheduler }                    Normal          Scheduled       Successfully assigned nginx-2932712833-91j0t to 127.0.0.1
  33s           33s             1       {kubelet 127.0.0.1}                     Normal          SandboxReceived Pod sandbox received, it will be created.
  6s            6s              1       {kubelet 127.0.0.1}                     Warning         FailedSync      Error syncing pod, skipping: failed to "CreatePodSandbox" for "nginx-2932712833-91j0t_default(2d3f6a43-df19-11e6-899a-507b9d4141fa)" with CreatePodSandboxError: "CreatePodSandbox for pod \"nginx-2932712833-91j0t_default(2d3f6a43-df19-11e6-899a-507b9d4141fa)\" failed: rpc error: code = 2 desc = reading pid from init pipe: EOF"
```

**Comment by sameo** [comment]
@runcom can you try enabling cc-proxy first:

```
systemctl start cc-proxy
```

**Comment by runcom** [comment]
> @runcom can you try enabling cc-proxy first:
> 
> systemctl start cc-proxy
> (It should start automatically upon socket activation, but it seems not to be the case)

that made the pause container to start up with clean containers. However I'm now seeing an issue when it can't actually start the "nginx" container because id is already reserved. Maybe a bug somewhere..

**Comment by runcom** [comment]
@sameo apparently I got a cc vm running with infra container but when it comes to starting the nginx container something goes wrong and cc-oci-runtime list shows it as stopped. Also, there  might be a bug there since the kubelet shows "created" and "started" while the status is still pending (not sure about networking in clear containers also).

**Comment by runcom** [comment]
> (It should start automatically upon socket activation, but it seems not to be the case)

@sameo the socket is disabled by default on installation btw.

**Comment by sameo** [comment]
> @sameo the socket is disabled by default on installation btw.

Yes, we're working on that one. At the moment, SELinux prevents us from enabling it by default.

**Comment by sameo** [comment]
> @sameo apparently I got a cc vm running with infra container but when it comes to starting the nginx container something goes wrong and cc-oci-runtime list shows it as stopped. Also, there might be a bug there since the kubelet shows "created" and "started" while the status is still pending (not sure about networking in clear containers also).

@runcom could you please share which commands you're using to start an nginx pod, so that I can try reproducing here ?

**Comment by runcom** [comment]
@sameo it works now! and it's great to see CC running with CRI-O :)

**Comment by rhatdan** [comment]
What SELinux issues are you seeing?

**Comment by runcom** [comment]
@rhatdan nothing related to SELinux - even if I tested it with setenforce 0 just to be sure - I'll re-test with selinux enabled and let you know.

**Comment by rhatdan** [comment]
@sameo @runcom showed me a demo, very nice.  We should be blogging and putting the demo up on youtube.  Nice to see alternate runtimes working under kubernetes using CRI-O.

**Comment by sameo** [comment]
@rhatdan Nothing related to SELinux itself, but the fact that we were not installing any TE file for our proxy socket. This should be fixed now with https://github.com/01org/cc-oci-runtime/pull/612

**Comment by sameo** [comment]
> @sameo @runcom showed me a demo, very nice. We should be blogging and putting the demo up on youtube. Nice to see alternate runtimes working under kubernetes using CRI-O.

It is very nice indeed :) It's been very easy to get the right changes merged into ocid, this is a truly open project.
Blogging about it would be great. Would the k8s blog be the right place ?

**Comment by rhatdan** [comment]
That would work fine.  Having a quote like " It's been very easy to get the right changes merged into ocid, this is a truly open project." would be awesome.

I will try to get a blog on a Red Hat site also.
